### PR TITLE
lib: add simpletests for internationalization effect

### PR DESCRIPTION
**Pull request by lisaw0**
add one simpletest that formats manually after Japanese-format and one with default effect

- [x] I have read and accept the [Tokiwa Software Fuzion Contributor Agreement](https://github.com/tokiwa-software/fuzion/blob/main/CONTRIBUTOR_LICENSE_AGREEMENT.md).

### CODE REVIEW

File: tests/i18n_default_test/i18n_default_test.fz [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+#
+# You should have received a copy of the GNU General Public License along with The
+# Fuzion language implementation.  If not, see <https://www.gnu.org/licenses/>.
+
+
+# -----------------------------------------------------------------------
+#
+#  Tokiwa Software GmbH, Germany
+#
+#  Source code of Fuzion test i18n_default_test2
```
Comment (michaellilltokiwa) [inline tests/i18n_default_test/i18n_default_test.fz:20]:
```suggestion
#  Source code of Fuzion test i18n_default_test
```

File: tests/i18n_test/i18n_test.fz [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+#
+#  Source code of Fuzion test i18n_test2
+#
+# -----------------------------------------------------------------------
+
+i18n_test is
+
+    #formates given date after Japanese-format
+    main ! internationalization.provide =>
+        say "{internationalization.provide.env.date (30) (5) (2024)}"Z
```
Comment (lisaw0) [inline tests/i18n_test/i18n_test.fz:28]:
```suggestion
        say "{internationalization.provide.env.date (30) (5) (2024)}"
```

File: tests/i18n_default_test/i18n_default_test.fz [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+#
+# -----------------------------------------------------------------------
+
+i18n_default_test is
+
+    #formates given date after default-setting, in this case German-format
+    main ! internationalization.provide =>
+        say "{internationalization.provide.current.date (30) (5) (2024)}"
+
+    main
```
Comment (simonvonhackewitz) [inline tests/i18n_default_test/i18n_default_test.fz:30]:
```suggestion
    h : envir.Vars_Handler is
      redef get(v String) option String => "de_DE.UTF-8"

    (envir.vars h).instate_self ()->

        #formates given date after default-setting, in this case German-format
        main ! internationalization.provide =>
            say "{internationalization.provide.current.date (30) (5) (2024)}"

        main
```

This uses a handler for the env.vars effect that returns always `de_DE.UTF-8`

File: tests/i18n_default_test/Makefile [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+# -----------------------------------------------------------------------
+#
+#  Tokiwa Software GmbH, Germany
+#
+#  Source code of Fuzion test Makefile
+#
+# -----------------------------------------------------------------------
+
+override NAME = i18n_default_test
+export LANG=de_DE.UTF-8
```
Comment (simonvonhackewitz) [inline tests/i18n_default_test/Makefile:25]:
```suggestion
```

This is not necessary then.

**Review by simonvonhackewitz — CHANGES_REQUESTED** [review]

**Review by michaellilltokiwa — APPROVED** [review]

File: tests/i18n_default_test/i18n_default_test.fz [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+i18n_default_test is
+
+    h : envir.Vars_Handler is
+      redef get(v String) option String => "de_DE.UTF-8"
+
+    (envir.vars h).instate_self ()->
+
+        #formates given date after default-setting, in this case German-format
+        main ! internationalization.provide =>
+            say "{internationalization.provide.current.date (30) (5) (2024)}"
```
Comment (michaellilltokiwa) [inline tests/i18n_default_test/i18n_default_test.fz:33]: Does this work?
```suggestion
            say "{internationalization.provide.current.date 30 5 2024}"
```

File: tests/i18n_test/i18n_test.fz [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+#
+#  Source code of Fuzion test i18n_test2
+#
+# -----------------------------------------------------------------------
+
+i18n_test is
+
+    #formates given date after Japanese-format
+    main ! internationalization.provide =>
+        say "{internationalization.provide.env.date (30) (5) (2024)}"
```
Comment (michaellilltokiwa) [inline tests/i18n_test/i18n_test.fz:28]: Does this work?
```suggestion
        say "{internationalization.provide.env.date 30 5 2024}"
```

**Review by michaellilltokiwa — APPROVED** [review]

**Review by michaellilltokiwa — APPROVED** [review]

**Pull request merged**